src/dc_msg.rs

```diff
@@ -269,87 +271,37 @@ pub unsafe fn dc_msg_empty(mut msg: *mut dc_msg_t) {
-                *ret_msgtype = Viewtype::Audio;
-                *ret_mime = dc_strdup(b"audio/mpeg\x00" as *const u8 as *const libc::c_char)
-            } else if strcmp(suffix, b"aac\x00" as *const u8 as *const libc::c_char) == 0i32 {
-                *ret_msgtype = Viewtype::Audio;
-                *ret_mime = dc_strdup(b"audio/aac\x00" as *const u8 as *const libc::c_char)
-            } else if strcmp(suffix, b"mp4\x00" as *const u8 as *const libc::c_char) == 0i32 {
-                *ret_msgtype = Viewtype::Video;
-                *ret_mime = dc_strdup(b"video/mp4\x00" as *const u8 as *const libc::c_char)
-            } else if strcmp(suffix, b"jpg\x00" as *const u8 as *const libc::c_char) == 0i32
-                || strcmp(suffix, b"jpeg\x00" as *const u8 as *const libc::c_char) == 0i32
-            {
-                *ret_msgtype = Viewtype::Image;
-                *ret_mime = dc_strdup(b"image/jpeg\x00" as *const u8 as *const libc::c_char)
-            } else if strcmp(suffix, b"png\x00" as *const u8 as *const libc::c_char) == 0i32 {
-                *ret_msgtype = Viewtype::Image;
-                *ret_mime = dc_strdup(b"image/png\x00" as *const u8 as *const libc::c_char)
-            } else if strcmp(suffix, b"webp\x00" as *const u8 as *const libc::c_char) == 0i32 {
-                *ret_msgtype = Viewtype::Image;
-                *ret_mime = dc_strdup(b"image/webp\x00" as *const u8 as *const libc::c_char)
-            } else if strcmp(suffix, b"gif\x00" as *const u8 as *const libc::c_char) == 0i32 {
-                *ret_msgtype = Viewtype::Gif;
-                *ret_mime = dc_strdup(b"image/gif\x00" as *const u8 as *const libc::c_char)
-            } else if strcmp(suffix, b"vcf\x00" as *const u8 as *const libc::c_char) == 0i32
-                || strcmp(suffix, b"vcard\x00" as *const u8 as *const libc::c_char) == 0i32
-            {
-                *ret_msgtype = Viewtype::File;
-                *ret_mime = dc_strdup(b"text/vcard\x00" as *const u8 as *const libc::c_char)
-            }
-        }
-    }
-    free(suffix as *mut libc::c_void);
-    free(dummy_buf as *mut libc::c_void);
+pub fn dc_msg_guess_msgtype_from_suffix(path: &Path) -> Option<(Viewtype, &str)> {
+    static KNOWN: phf::Map<&'static str, (Viewtype, &'static str)> = phf_map! {
+        "mp3"   => (Viewtype::Audio, "audio/mpeg"),
+        "aac"   => (Viewtype::Audio, "audio/aac"),
+        "mp4"   => (Viewtype::Video, "video/mp4"),
+        "jpg"   => (Viewtype::Image, "image/jpeg"),
+        "jpeg"  => (Viewtype::Image, "image/jpeg"),
+        "png"   => (Viewtype::Image, "image/png"),
+        "webp"  => (Viewtype::Image, "image/webp"),
+        "gif"   => (Viewtype::Gif,   "image/gif"),
+        "vcf"   => (Viewtype::File,  "text/vcard"),
+        "vcard" => (Viewtype::File,  "text/vcard"),
+    };
+    let extension = path.extension()?.to_str()?;
+
+    KNOWN.get(extension).map(|x| *x)
```

Comment:
the original dc_get_filesuffix_lc function uses dc_strlower() on the extension.  i guess this needs to be happen here as well.

Reply:
Correct. Pushed one more commit.

By the way, any ideas why type signature is required?